File: market/test/java/ru/yandex/market/pers/shopinfo/controller/PagematchControllerTest.java
package ru.yandex.market.pers.shopinfo.controller;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.http.ResponseEntity;

import ru.yandex.market.common.test.spring.FunctionalTestHelper;
import ru.yandex.market.pers.shopinfo.test.context.FunctionalTest;

/**
 * Тест на работоспособность метода shopinfo /pagematch.
 *
 * @author stani on 05.03.18.
 */
public class PagematchControllerTest extends FunctionalTest {

    @Test
    public void testPagematch() {
        ResponseEntity<String> response = FunctionalTestHelper.get(urlBasePrefix + "/pagematch");
        Assertions.assertTrue(response.getBody().split("\n").length > 1);
    }
}

File: market/test/kotlin/ru/yandex/market/mbi/orderservice/api/controller/PageMatchControllerTest.kt
package ru.yandex.market.mbi.orderservice.api.controller

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.http.HttpStatus
import ru.yandex.market.common.test.spring.FunctionalTestHelper
import ru.yandex.market.mbi.orderservice.api.ApiUrl
import ru.yandex.market.mbi.orderservice.api.FunctionalTest

/**
 * Тесты для [PageMatchController].
 */
internal class PageMatchControllerTest : FunctionalTest() {

    private val url: String by ApiUrl("/pagematch")

    @Test
    fun `test pagematch format`() {
        val response = FunctionalTestHelper.get(url)

        assertThat(response.statusCode)
            .isEqualTo(HttpStatus.OK)
        assertThat(response.body)
            .contains("ping\t/ping\torder-service")
    }
}
